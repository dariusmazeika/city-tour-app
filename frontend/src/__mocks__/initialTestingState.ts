export default {
  localization: {
    currentLanguage: 'en',
  },
};
