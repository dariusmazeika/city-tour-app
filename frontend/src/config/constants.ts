export enum Forms {
    loginForm = 'login-form',
}
