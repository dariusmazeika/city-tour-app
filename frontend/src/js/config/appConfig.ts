import { getCookie } from '../utils/cookies';
import { Cookies } from './constants';

export function configValue(param: string) {
  return window._app_conf[param] || param;
}
export function getDefaultLanguage() {
  const cookieLanguage = getCookie(Cookies.defaultLang);
  return cookieLanguage || configValue('default_language');
}

export function getMessageKeyTranslation(msgid: string, language: string) {
  if (!language) {
    return msgid;
  }
  const languageMessages = window._app_messages[language] || {};
  return languageMessages[msgid] || msgid;
}
