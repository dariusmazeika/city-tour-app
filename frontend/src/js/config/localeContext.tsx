import * as React from 'react';

import { getDefaultLanguage } from './appConfig';

export interface LocaleContextType {
  localeContext: string;
}
const localeContext = React.createContext(getDefaultLanguage());

/* tslint:disable: variable-name */
export const AppContextProvider = localeContext.Provider;

export const AppContextConsumer = localeContext.Consumer;

export const withLocaleContext = (Component: any) => {
  return function ComponentBoundWithAppContext(props: any) {
    return (
      <AppContextConsumer>
        {appContext => <Component {...props} localeContext={appContext} />}
      </AppContextConsumer>
    );
  };

};
/* tslint:enable */
