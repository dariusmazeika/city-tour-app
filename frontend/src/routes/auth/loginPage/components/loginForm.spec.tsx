import * as React from 'react';
import { mount, shallow, render } from 'enzyme';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import LoginForm from './loginForm';
import configureMockStore from 'redux-mock-store';
import { Forms } from '../../../../config/constants';
import initialTestingState from '../../../../__mocks__/initialTestingState';

describe('<LoginForm />', () => {
  let store;
  beforeAll(() => {
    store = configureMockStore([])({
      ...initialTestingState,
      form: {
        [Forms.loginForm]: {
          values: {
          },
        },
      }});
  });
  it('should match snapshot', () => {
    const onSubmitMock = jest.fn();
    const wrapper = create(<Provider store={store}>
            <LoginForm  onSubmit={onSubmitMock}/>
        </Provider>);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should simulate click', () => {
    const onSubmitMock = jest.fn();
    const wrapper = shallow(<Provider store={store}>
            <LoginForm  onSubmit={onSubmitMock}/>
        </Provider>);
    expect(wrapper.find(LoginForm).length).toEqual(1);
    wrapper.find(LoginForm).simulate('submit');
    expect(onSubmitMock).toBeCalled();
  });
});
