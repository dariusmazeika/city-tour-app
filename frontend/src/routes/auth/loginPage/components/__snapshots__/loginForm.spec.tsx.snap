// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should match snapshot 1`] = `
<form
  className="form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="login-form"
  >
    <div
      className="form-field"
    >
      <div
        className="form-input"
      >
        <input
          autoComplete="off"
          className=""
          disabled={false}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="msg_label_email"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      className="form-field"
    >
      <div
        className="form-input"
      >
        <input
          autoComplete="off"
          className=""
          disabled={false}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="msg_label_password"
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <button
    type="submit"
  >
    Save
  </button>
</form>
`;
